web/src/typeahead_helper.ts

```diff
@@ -672,3 +673,11 @@ export function query_matches_name(
 ): boolean {
     return typeahead.query_matches_string_in_order(query, user_group_or_stream.name, " ");
 }
+
+export function query_matches_group_name(query: string, user_group: UserGroupPillData): boolean {
+    return typeahead.query_matches_string_in_order(
```

Comment:
There are some places where we still use `query_matches_name` for groups, but that would work as those typeahead does not include system groups and the only thing that is different in this is that it considers `display_name` while matching with system groups.

Reply:
Yeah, that seems OK, though it would be better to just have them use this too, to avoid problems with cargo-culting creating broken code.

Reply:
Added a commit for this.